starlite/utils/compat.py

```diff
@@ -19,3 +20,23 @@ async def async_next(gen: AsyncGenerator[T, Any], default: D | EmptyType = Empty
             if default is not Empty:
                 return default  # type: ignore[return-value]
             raise exc
+
+
+C = TypeVar("C", bound=Callable)
+
+
+def validate_arguments(fn: C) -> C:
+    """Proxy pydantic `validate_arguments` decorator to ignore python 3.8, which doesn't support future
+    annotations.
+
+    :param fn: A callable to decorate.
+    :return: Either the decorated callable, if python version is 3.8, or the callable itself.
+    """
+    if version_info < (3, 9):  # pragma: no cover
```

Comment:
So we just don't validate the arguments for `3.8`? That should definitely be mentioned in the docs somewhere then, since it could cause bugs if you rely on the validation